code/game/objects/items/devices/radio/headset.dm

```diff
@@ -164,20 +171,29 @@
 	translate_hive = FALSE
 	syndie = FALSE
 
+	tracking_options = length(inbuilt_tracking_options) ? inbuilt_tracking_options.Copy() : list()
 	for(var/i in keys)
 		var/obj/item/device/encryptionkey/key = i
 		for(var/ch_name in key.channels)
 			if(ch_name in channels)
 				continue
 			channels += ch_name
 			channels[ch_name] = key.channels[ch_name]
+		for(var/tracking_option in key.tracking_options)
+			tracking_options[tracking_option] = key.tracking_options[tracking_option]
 		if(key.translate_binary)
 			translate_binary = TRUE
 		if(key.translate_hive)
 			translate_hive = TRUE
 		if(key.syndie)
 			syndie = TRUE
 
+	var/list/tracking_stuff = list()
+	for(var/thing in tracking_options)
+		tracking_stuff += tracking_options[thing]
+	if(!(locate_setting in tracking_stuff))
+		locate_setting = tracking_stuff[1]
```

Comment:
If I'm reading this correctly I believe the following:
If a headset has only the tracker from a key and that key is removed the locate_setting will not be in tracking_stuff and tracking_stuff will be empty and throw a runtime.

Probably needs a check for if there are any tracking options. Also, these variable names could probably be better :eyes:

Let me know if I'm misreading.